stats/stats.go

```diff
@@ -135,11 +135,13 @@ var (
 		2000, 3000, 4000, 5000, 7500, 10000, // another order of magnitude coarsly covered
 		20000, 30000, 40000, 50000, 75000, 100000, // ditto, the end
 	}
-	numValues  = len(histogramBucketValues)
-	numBuckets = numValues + 1 // 1 special first bucket is <= 0; and 1 extra last bucket is > 100000
-	firstValue = float64(histogramBucketValues[0])
-	lastValue  = float64(histogramBucketValues[numValues-1])
-	val2Bucket []int
+	numValues            = len(histogramBucketValues)
+	numBuckets           = numValues + 1 // 1 special first bucket is <= 0; and 1 extra last bucket is > 100000
+	firstValue           = float64(histogramBucketValues[0])
+	lastValue            = float64(histogramBucketValues[numValues-1])
+	maxArrayValue        = int32(1000) // Last value looked up as O(1) array, the rest is linear search
+	val2Bucket           []int         //up to 1000
+	linearTimeVal2Bucket []int         // later than 1000 to 100000
```

Comment:
I don't think this is needed, you just need maxArrayValueIndex and use the existing histogramBucketValues for the search